### PR TITLE
test: migrate to hardhat

### PR DESCRIPTION
**Pull request by deanamiel**
Migrated tests to hardhat with the exception of GeneralMessagePassing.js due to errors. Once all test files are migrated away from waffle, hardhat-toolbox can be installed to replace hardhat-waffle within the hardhat config file.

### CODE REVIEW

File: test/gmp/GeneralMessagePassing.js [hunk 1]
```diff
@@ -5,14 +5,14 @@ const {
     Contract,
     utils: { defaultAbiCoder, arrayify, keccak256 },
 } = require('ethers');
+const { ethers } = require('hardhat');
 const { deployContract, MockProvider, solidity } = require('ethereum-waffle');
```
[the remark this answers is not in the document]
Comment (deanamiel) [reply test/gmp/GeneralMessagePassing.js:9]: I can, the only issue is that it throws a lint error when I reference ethers.constants.AddressZero if I remove it

[the remark this answers is not in the document]
Comment (milapsheth) [reply test/gmp/GeneralMessagePassing.js:9]: Yeah, I was actually referring to `const { deployContract, MockProvider, solidity } = require('ethereum-waffle');`. Note the gh adds a comment just under the line being highlighted, although the context makes it seem like the comment was on the line before that

[the remark this answers is not in the document]
Comment (deanamiel) [reply test/gmp/GeneralMessagePassing.js:9]: Removed all require('ethers'); and replaced with const { ethers } = require('hardhat')

File: test/AxelarGateway.js [hunk 2]
```diff
@@ -353,10 +352,10 @@ describe('AxelarGateway', () => {
...
             );
-            await expect(gateway.execute(secondInput))
-                .to.not.emit(gateway, 'TokenDeployed')
-                .and.to.emit(gateway, 'Executed')
-                .withArgs(secondCommandID);
+            const executeTx = await gateway.execute(secondInput);
+
+            await expect(executeTx).to.not.emit(gateway, 'Executed');
+            await expect(executeTx).to.not.emit(gateway, 'TokenDeployed');
         });
```
Comment (deanamiel) [inline test/AxelarGateway.js:359]: @milapsheth check out this fix that Kiryl and I made

**Review by re1ro — APPROVED** [review]

**Pull request merged**